pkg/kubelet/server/server.go

```diff
@@ -185,9 +194,12 @@ func ListenAndServePodResources(socket string, podsProvider podresources.PodsPro
 	podresourcesapi.RegisterPodResourcesListerServer(server, podresources.NewV1PodResourcesServer(podsProvider, devicesProvider, cpusProvider))
 	l, err := util.CreateListener(socket)
 	if err != nil {
-		klog.Fatalf("Failed to create listener for podResources endpoint: %v", err)
+		klog.ErrorS(err, "Failed to create listener for podResources endpoint")
```

Comment:
When replacing `Fatalf`  we should add `os.Exit(1)` to maintain exit. Same as you did above when replacing `Fatal`